transaction/transaction/src/main/java/io/helidon/transaction/Tx.java

```diff
@@ -0,0 +1,303 @@
+        Services.get(TxSupport.class)
+                .transaction(type,
+                             () -> {
+                                 task.run();
+                                 return null;
+                             });
+    }
+
+/* Removed: need to decide whether this is needed at all
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is finished manually. Task computes and returns result.
+     *
+     * @param task task to run in transaction
+     * @param <T>  the result type of the task
+     * @return computed task result
+     * /
+    public static <T> T transaction(Function<Transaction, T> task) {
+        return transaction(Type.REQUIRED, task);
+    }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is finished manually. Task computes and returns result.
+     *
+     * @param type transaction type
+     * @param task task to run in transaction
+     * @param <T>  the result type of the task
+     * @return computed task result
+     * /
+    public static <T> T transaction(Type type, Function<Transaction, T> task) {
+        return Services.get(TxSupport.class)
+                .transaction(type, task);
+    }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task does not return any result.
+     *
+     * @param task task to run in transaction
+     * /
+    public static void transaction(Consumer<Transaction> task) {
+        transaction(Type.REQUIRED, task);
+    }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task does not return any result.
+     *
+     * @param type transaction type
+     * @param task task to run in transaction
+     * /
+    public static void transaction(Type type, Consumer<Transaction> task) {
+        Services.get(TxSupport.class)
+                .transaction(type,
+                             t -> {
+                                 task.accept(t);
+                                 return null;
+                             });
+    }
+
+    /**
+     * Start transaction.
+     * Transaction is finished manually.
+     *
+     * @return transaction handler
+     * /
+    public static Transaction transaction() {
+        return transaction(Type.REQUIRED);
+    }
+
+    /**
+     * Start transaction.
+     * Transaction is finished manually.
+     *
+     * @param type transaction type
+     * @return transaction handler
+     * /
+    public static Transaction transaction(Type type) {
+        return Services.get(TxSupport.class)
+                .transaction(type);
+    }
+
+*/
+
+    /**
+     * The {@link Transaction} interface defines methods that allow user code to manage transaction boundaries.
+     */
+    public interface Transaction {
+
+        /**
+         * Complete current transaction.
```

Comment:
…or throw some kind of exception? What if I call it twice in a row? Is there a state diagram? and so on

Reply:
`Transaction` interface was removed.